### DISCUSSION TITLE
DA rankability discussion

### DISCUSSION THREAD

**Discussion by Givikap120** [ppy/osu#27799] [Ideas]
Here I want to analyze possibility to rank different aspects of DA in the current state of pp system. Open for other opinions.
- CS increase. Potential abuse - high CS bonus. Some farm maps can be overweight with usage of this bonus, but this overweightness is minor in comparison to already existing problems. My opinion - rankable (abuseness = 20%)
- CS decrease. Potential abuse - d/t^2 abuse. Some maps despite decreased pp award - can be overweight to some extent with decreased CS. The most notable example - sunglow DT, can be considerably overweight with CS2. My opinion - rankable (abuseness = 30%)
- HP increase. Makes map harder to pass without affecting pp. My opinion - rankable (abuseness = 0%)
- HP decrease. Makes map easier to pass without affecting pp. Considering that pp doesn't awards for passing - this won't change anything. My opinion - rankable (abuseness = 0%)
- OD increase. Allowing players with skill to get very low UR on simple rhythm maps to farm big amount of pp. Considering that most of these maps (with simple rhythm) usually have already high OD - this will not change very much. My opinion - rankable (abuseness = 20%).
- OD decrease. Absolutely free pp on maps with high OD for many players (most notable examples - DT speed maps). Can allow raketapping, doubletapping and mashing to achieve extremely high pp scores. DO NOT rank this until pp solution to this (like stat acc) will be ready. My opinion - unrankable (abuseness = 90%)
- AR increase. First usage for this is on gimmicky low AR maps. Considering that those maps are already unfarmable, changing AR will make them at least playable to average farmer, so this will not change anything. Another usecase - abusing high AR bonus on high end. High AR bonus is **very** meta on high end (top100) level now, and ability to set every map to something like AR10.7 or AR10.8 will cause some pp inflation on high end. Also, setting AR to very high values (>11) can lead to very big high AR multiplier, abusable with memorization. My opinion - rankable if fix high AR bonuses on very high AR (abuseness = 40%)
- AR decrease. Considering that low AR is pretty much underweight now - usually AR decrease will only be used to non-DT players to play DT maps without high AR reading abilities. However, considering that it's possible to set AR to -10, this opens possibility to claim very high low AR bonus with memorization. This means that maps like Because Maybe p3 +HTFL can be overweight with AR-10. My opinion - rankable if unrank AR<0 (abuseness = 35%)

**Comment by Stoppedpuma** [comment]
I'm 90% sure it would be more beneficial to decide on multipliers before making it ranked since **DA needs to match the 0.06x (0.12x if playing catch) multiplier when using HR values and mostly match the EZ multiplier when using its values (minus extra lives part).** Here are some values I think could work for osu!:

CS: 0.03x per 30% increase, -0.07x per 50% decrease
AR: 0.01x per 40% increase, -0.03x per 50% decrease
OD: 0.02x per 40% increase, -0.1x per 50% decrease
HP: shouldn't increase multiplier, -0.1x per 50% decrease

Notes:
The missing -0.2x with easy values is due to not having extra lives.
"per" doesn't mean only adjust at x% increase/decrease, it's just to show what parts of HR and EZ affect multipliers using their percentages.
My suggested values are for osu! as I don't play other modes enough to give any valuable input towards multipliers in their case.

**Reply by Avelia-tr** [reply]
first : your formula doesn't work for map where a stat is equal to 0

for CS : this would lead to a 0.93 multiplying for having decreased to 5 on a cs10 map. This is wayy too low since it marginalize those map. A correct multiplier would look more like 0.3 or even lower. 

Increasing multiplier for increasing AR is stupid for a lot of reason, the main one being that this would make all leaderboard on low ar map irrelevant wich is a really bad thing. I think the inverse could maybe work ? (as in -0.1 for every 40% increase) 

OD should be calibrated too the ur (like wich ur you'd need to match X% of acc ) or simply not have a multiplier for increasing it

Overall I think it should be more the ez and hr multiplier that changes to adapt to DA and not the inverse, those (especially EZ) Are mainly agreed to be really flawed by the player base. It would just lead to the same error. 

The main goals imo should be : 
Preventing cheese on low ar/gimmick map
Allowing niche score a bit of spotlight (ie an AR-10 doesn't really deserve any nerf, same a cs 11 score deserve a really good multiplier ) 

Like you I'm talking only about osu as I don't know for other game mode

**Reply by Stoppedpuma** [reply]
Have to agree with most of what you're saying, these values were mostly thought of within 10 minutes while I was making the comment. It was meant to define which stats of HR and EZ give multiplier to the mod in order to match the multiplier value of that mod when using its stat values in DA.

Regarding AR: I agree that increasing AR on low AR maps shouldn't increase multiplier. On the opposite side, I think that a large percent of players (myself included) see maps above AR 10 and near AR 11 harder than if they were AR 10, maybe a multiplier increase for above AR 10.5 would be a good idea?

**Reply by Avelia-tr** [reply]
I don't think it's worth sacrifying gimmick maps for that but it could work yep. 

I do think a multiplier based on 'modified stat - base stat' could work really well and that sadly not compatible with that.

**Comment by sz-da** [comment]
HP does actually affect pp and difficulty for blinds, so there is a possibility of abusing blinds with high HP. Not sure how abusable it is though.

**Reply by Givikap120** [reply]
Blinds pp is using HP in it's calculation
How good it is - it's a different question
But if you gonna FC the map - I don't think increasing HP will help you that much (at least in non-CL HP system 100%)
But yea - Blinds is fking ruining everything, who thought that make mod that based on HP is a good idea

**Comment by Rekunan** [comment]
I think that CS and AR changes are more abusable than you think, mainly cause of CS cap and AR bonuses respectively.

**Reply by Givikap120** [reply]
1) How is CS cap makes it more abusable?
2) I already covered all AR bonuses cases and how they can be abused

**Reply by Rekunan** [reply]
1. Cause it means you can lower maps above cs cap to get same pp for lower difficulty
2. That one's mb, I meant to say I disagree with this part:
> First usage for this is on gimmicky low AR maps. Considering that those maps are already unfarmable, changing AR will make them at least playable to average farmer, so this will not change anything.

I think this will actually be a big change, so we def need reading pp.

**Reply by Givikap120** [reply]
Both points is "changing severely underweight maps to just underweight". Why is it so hard to understand that changing AR8 map to AR9.5 isn't nowhere near being "abusable", "overweight", "broken", "unrankable" or "unfair", same with CS9 to CS6.5

First statement is just blatantly wrong, because it's not "same pp" but "same high CS bonus", what isn't makes map overweight at all, ESPECIALLY when considering how CS>7 maps are usually structured
Gimmick maps are underweight not only because low AR, and even if they was underweight only because of low AR - changing it will just make them as farm as normal farm maps
